### PR TITLE
support for endport

### PR DESCRIPTION
**Pull request by haim-kermany**

### CODE REVIEW

File: tests/example_policies/withEndPort/withEndPort-scheme.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - np2
+    expected: 1
+  - name: emptiness
+    emptiness:
+      - np1
+    expected: 0
+  - name: equivalence
+    equivalence:
+      - np1
+      - np2
```
Comment (zivn) [inline tests/example_policies/withEndPort/withEndPort-scheme.yaml:30]: an expected field is missing in the equivalence query

File: tests/example_policies/withEndPort/withEndPort-scheme.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  - name: np2
+    networkPolicyList:
+      - withEndPort-networkpolicy2.yaml
+    expected_warnings: 0
+
+queries:
+  - name: redundnacy_np1
+    redundancy:
+      - np1
+    expected: 0
```
Comment (zivn) [inline tests/example_policies/withEndPort/withEndPort-scheme.yaml:18]: What does this query check?

File: tests/example_policies/withEndPort/withEndPort-scheme.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  - name: redundnacy_np2
+    redundancy:
+      - np2
+    expected: 1
+  - name: emptiness
+    emptiness:
+      - np1
+    expected: 0
```
Comment (zivn) [inline tests/example_policies/withEndPort/withEndPort-scheme.yaml:26]: What does this query check? Is it needed?

File: network-config-analyzer/K8sPolicyYamlParser.py [hunk 4]
```diff
@@ -153,18 +153,28 @@ def parse_port(self, port):
...
+        if port_id and endport_id:
+            if isinstance(port_id, str):
+                self.syntax_error("If Port is a string (named port) EndPort cannot be specified", port)
+            if port_id > endport_id:
+                self.syntax_error("endPort must be equal or bigger than port", port)
+            dest_port_set.add_port_range(port_id, endport_id)
+        elif port_id:
             dest_port_set.add_port(port_id)
+        elif endport_id:
+            self.syntax_error("If an EndPort is specified a Port must also be specified", port)
```
Comment (zivn) [inline network-config-analyzer/K8sPolicyYamlParser.py:176]:
```suggestion
            self.syntax_error('endPort cannot be specified without Port being specified', port)
```

File: network-config-analyzer/K8sPolicyYamlParser.py [hunk 5]
```diff
@@ -153,18 +153,28 @@ def parse_port(self, port):
...
             protocol = 'TCP'
         if protocol not in ['TCP', 'UDP', 'SCTP']:
             self.syntax_error('protocol must be "TCP" or "UDP" or "SCTP"', port)
 
         res = ConnectionSet()
         dest_port_set = PortSet(port_id is None)
-        if port_id:
+        if port_id and endport_id:
+            if isinstance(port_id, str):
+                self.syntax_error("If Port is a string (named port) EndPort cannot be specified", port)
```
Comment (zivn) [inline network-config-analyzer/K8sPolicyYamlParser.py:169]:
```suggestion
                self.syntax_error('endPort cannot be specified when port is a string (named port)', port)
```

File: network-config-analyzer/K8sPolicyYamlParser.py [hunk 6]
```diff
@@ -153,18 +153,28 @@ def parse_port(self, port):
...
+            if port_id > endport_id:
+                self.syntax_error("endPort must be equal or bigger than port", port)
```
Comment (zivn) [inline network-config-analyzer/K8sPolicyYamlParser.py:171]:
```suggestion
                self.syntax_error('endPort must be equal or larger than port', port)
```

File: network-config-analyzer/K8sPolicyYamlParser.py [hunk 7]
```diff
@@ -153,18 +153,28 @@ def parse_port(self, port):
...
         if protocol not in ['TCP', 'UDP', 'SCTP']:
             self.syntax_error('protocol must be "TCP" or "UDP" or "SCTP"', port)
 
         res = ConnectionSet()
         dest_port_set = PortSet(port_id is None)
-        if port_id:
+        if port_id and endport_id:
+            if isinstance(port_id, str):
+                self.syntax_error("If Port is a string (named port) EndPort cannot be specified", port)
+            if port_id > endport_id:
```
Comment (zivn) [inline network-config-analyzer/K8sPolicyYamlParser.py:170]: Should we give a warning if port_id == endport_id (in which case endPort is redundant)

**Review by zivn — CHANGES_REQUESTED** [review]

File: network-config-analyzer/K8sPolicyYamlParser.py (hunk 7 above)
Comment (haim-kermany) [reply network-config-analyzer/K8sPolicyYamlParser.py:170]: NO way!!!

**Review by zivnevo — APPROVED** [review]

**Pull request merged**